app/merchant/views.py

```diff
@@ -14,31 +14,52 @@ def index():
 
 
 @merchant.route('/view/all')
+@merchant.route('/view/all/<int:page>')
 @login_required
 @merchant_required
-def listing_view_all():
+def listing_view_all(page=1):
     """Search for listings"""
-    main_search_term = request.args.get('main-search', "", type=str)
+    main_search_term = request.args.get('mainsearch', "", type=str)
     favorite = True if request.args.get('favorite') == "on" else False
-    name_search_term = request.args.get('name-search', "", type=str)
+    sortby = request.args.get('sortby', "", type=str)
+    name_search_term = request.args.get('namesearch', "", type=str)
     min_price = request.args.get('min-price', "", type=float)
     max_price = request.args.get('max-price', "", type=float)
-    listings = Listing.search(available=True,
-                              favorite=favorite,
-                              min_price=min_price,
-                              max_price=max_price,
-                              name_search_term=name_search_term,
-                              main_search_term=main_search_term)
-
-    return render_template('merchant/view_listings.html',
-                           listings=listings,
-                           main_search_term=main_search_term,
-                           min_price=min_price,
-                           max_price=max_price,
-                           name_search_term=name_search_term,
-                           favorite=favorite,
-                           cart_listings=current_user.get_cart_listings(),
-                           header="All listings")
+    category_search = request.args.get('category-search', "", type=str)
+    search = request.args.get('search', "", type=str)
```

Comment:
This was used to fix the following bug. When a user browses search results and decides to search on a page other than page 1, the pagination does not kick them back to page 1. this can be problematic if a user is say on Page 4, but the query the search only has 2 pages of results. The search arg helps prevent this.